### ISSUE TITLE
Nested Plugins

### ISSUE THREAD

**Issue by CooCooCaCha** [free-audio/clap#101]
In Bitwig, some built-in devices allow you to nest plugins inside them. For example, the delay effect allows you to insert as many plugins as you want inside the feedback signal path. 

It would be really cool if CLAP plugins could do this too.

**Comment by abique** [comment]
This is an interesting idea.

**Comment by baconpaul** [comment]
yeah i think channel strips, plugins-in-plugins, interaction with the host. These are all interesting ideas

I think there's two extensions I have in mind from some of these convos

1. Channel Strip - lets you know who is downstream of you and target them with polymer
2. Host-as-Factory - you can ask the host to create an instance of a plugin for you to manage subordinate to yourself

I definitely am going to take a swing at a proposal for the first but probably not for a few weeks.

**Comment by lorenzogatti** [comment]
Would it be possible for a plugin to specify that some of its output and input ports are meant to be used as the plugin's own send and return ports and optionally populated with "private" plugins? It would allow a suitable host GUI to choose such plugins, without abusing the standard structure of tracks, channels etc. to find a place for them.

A more advanced possibility: plugins asking the host to assign to them a variable number of ports. 
Example: generalizing an equalizer or multiband compressor to a plugin consisting of a crossover filter bank that divides the input into a variable number of bands (user-defined) and passes each through a different plugin (or plugin chain) before summing the results. 
I only see extensions that allow the host to ask the plugin to change port configuration, not the opposite.

**Comment by Reaper10** [comment]
would this support DSSI/LADSPA or vst?

**Comment by baconpaul** [comment]
Well not much has happened in this idea but no if I coded it up I would just code for to the clap api

**Comment by geraintluff** [comment]
I vote to close this issue, then
